.github/workflows/lint.yml

```diff
@@ -0,0 +1,36 @@
+name: Lint
+
+on:
+  push:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  run-linters:
+    name: Run linters
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [lts/hydrogen]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "pnpm"
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Run linter
+        run: pnpm run lint
```

Comment:
I usually have a changed script in package json like this:
`prettier --check . && eslint . --ext js,ts && tsc --noEmit`

Actually, it's not just linting, but I like to check the code format as well.